src/solvers/lowering/byte_operators.cpp

```diff
@@ -21,6 +21,46 @@ Author: Daniel Kroening, kroening@kroening.com
 #include <util/simplify_expr.h>
 #include <util/string_constant.h>
 
+/// Determine the member of maximum fixed bit width in a union type. If no
+/// member, or no member of fixed and non-zero width can be found, return
+/// nullopt.
+/// \param union_expr: Expression to use as compound operand in the resulting
+///   member_exprt.
+/// \param union_type: Type to determine the member of.
+/// \param ns: Namespace to resolve tag types.
+/// \return Pair of a member_exprt pointing to the maximum fixed bit-width
+///   member of \p union_type and the bit width of that member.
+static optionalt<std::pair<member_exprt, mp_integer>>
+find_maximum_union_component(
+  const exprt &union_expr,
+  const union_typet &union_type,
+  const namespacet &ns)
+{
+  const union_typet::componentst &components = union_type.components();
+
+  mp_integer max_width = 0;
+  typet max_comp_type;
+  irep_idt max_comp_name;
+
+  for(const auto &comp : components)
+  {
+    auto element_width = pointer_offset_bits(comp.type(), ns);
+
+    if(!element_width.has_value() || *element_width <= max_width)
+      continue;
+
+    max_width = *element_width;
+    max_comp_type = comp.type();
+    max_comp_name = comp.get_name();
+  }
+
+  if(max_width == 0)
+    return {};
+  else
+    return std::make_pair(
+      member_exprt{union_expr, max_comp_name, max_comp_type}, max_width);
```

Comment:
A bit surprising that a utility would make a member-expression, rather than just returning the widest component

Reply:
Good call - this makes one use slightly more work (having to construct the `member_exprt` in place), but avoids a parameter that isn't really necessary and could be used in a wrong way.